internal/handler/url.go

```diff
@@ -1,32 +1,30 @@
 package handler
 
 import (
-	"fmt"
+	"errors"
 	"github.com/delyke/urlShortener/internal/config"
 	"github.com/delyke/urlShortener/internal/service"
 	"github.com/go-chi/chi/v5"
 	"io"
+	"log"
 	"net/http"
+	"net/url"
 	"strings"
 )
 
 type Handler struct {
-	service service.ShortenURLService
+	service ShortenURLService
 }
 
-func NewHandler(service service.ShortenURLService) *Handler {
+func NewHandler(service ShortenURLService) *Handler {
 	return &Handler{service: service}
 }
 
 func (h *Handler) HandlePost(w http.ResponseWriter, r *http.Request) {
-	if r.Method != http.MethodPost {
-		w.WriteHeader(http.StatusMethodNotAllowed)
-		return
-	}
-
 	body, err := io.ReadAll(r.Body)
 	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Println(err)
```

Comment:
К ошибкам всегда лучше добавлять информацию о том, что пошло не так

```suggestion
		log.Printf("failed to read the request body: %v", err)
```